pkg/didcomm/protocol/transport/http/http_transport.go

```diff
@@ -0,0 +1,262 @@
+	resp, err := cs.client.Post(url, commContentType, bytes.NewBuffer([]byte(data)))
+	if err != nil {
+		log.Printf("HTTP Transport - Error posting did envelope to agent at [%s]: %v", url, err)
+		return "", err
+	}
+
+	var respData string
+	if resp != nil {
+		isStatusSuccess := resp.StatusCode == http.StatusAccepted || resp.StatusCode == http.StatusOK
+		if !isStatusSuccess {
+			return "", errors.Errorf("Warning - Received non success POST HTTP status from agent at [%s]: status : %v", url, resp.Status)
+		}
+		// handle response
+		defer func() {
+			err := resp.Body.Close()
+			if err != nil {
+				log.Printf("HTTP Transport - Error closing response body: %v", err)
+			}
+		}()
+		buf := new(bytes.Buffer)
+		_, err := buf.ReadFrom(resp.Body)
+		if err != nil {
+			return "", err
+		}
+		respData = buf.String()
+	}
+	return respData, nil
+}
+
+// creates a new instance of HTTP transport as a client
+func newHTTPClient(cfg *OutboundCommConfig) (*http.Client, error) {
+	var err error
+	var caCertPool tlsCertPool.CertPool
+	if cfg.CACertsPaths != "" {
+		caCertPool, err = tlsCertPool.NewCertPool(false)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to create new Cert Pool")
+		}
+
+		caCertsPaths := strings.Split(cfg.CACertsPaths, ",")
+		var caCerts []string
+		for _, path := range caCertsPaths {
+			if path == "" {
+				continue
+			}
+			// Create a pool with server certificates
+			caCert, e := ioutil.ReadFile(filepath.Clean(path))
+			if e != nil {
+				return nil, errors.Wrap(e, "Failed Reading server certificate")
+			}
+			caCerts = append(caCerts, string(caCert))
+		}
+
+		caCertPool.Add(tlsCertPool.DecodeCerts(caCerts)...)
+	} else {
+		caCertPool, err = tlsCertPool.NewCertPool(true)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// update the config's caCertPool
+	cfg.caCertPool = caCertPool
+
+	tlsConfig, err := buildNewCertPool(cfg.caCertPool)
+	if err != nil {
+		log.Printf("HTTP Transport - Failed to build/get Cert Pool: %s", err)
+		return nil, err
+	}
+
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+		Timeout: cfg.Timeout,
+	}, nil
+}
+
+func buildNewCertPool(tlsCertPool tlsCertPool.CertPool) (*tls.Config, error) {
+	cp, err := tlsCertPool.Get()
+	if err != nil {
+		return nil, err
+	}
+	// Create TLS configuration
+	// add other agents (server) certs here.. for now we trust CA certs
+	// Since this config cannot be manipulated once the client is created,
+	// we must find a way to trust other agents here (or only trust root CAs).
+	tlsConfig := &tls.Config{
+		// if RootCAs is nil, client will use the host's root CA instead
+		RootCAs:      cp,
+		Certificates: nil,
+	}
+	return tlsConfig, nil
+}
+
+// DIDCommRequestHandler will create a new handler to enforce Did-Comm HTTP transport specs
+// then routes processing to the passed in handler argument
+func DIDCommRequestHandler(handler http.Handler, commHandler *transport.DIDCommHandler) http.Handler {
```

Comment:
Normally, we would use an option pattern to pass one or more handlers.

Reply:
Inbound request handling will be added in a separate PR